controller/backup_target_controller.go

```diff
@@ -367,6 +430,43 @@ func (btc *BackupTargetController) reconcile(name string) (err error) {
 		return err
 	}
 
+	// Check if it is the default backup target or not.
+	backupTargetMap, err := btc.ds.ListBackupTargets()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get all backup targets")
+		return err
+	}
+	if len(backupTargetMap) == 1 {
+		if _, ok := backupTargetMap[name]; ok {
+			backupTarget.Spec.Default = true
+			backupTarget.Status.Default = true
+		}
+	} else {
+		if backupTarget.Spec.Default {
+			for _, btObj := range backupTargetMap {
+				if btObj.Status.Default && btObj.Name != name {
+					btObj.Status.Default = false
+					if _, err := btc.ds.UpdateBackupTargetStatus(btObj); err != nil {
+						log.WithError(err).Errorf("Failed to update backup target %s status", btObj.Name)
+						return err
+					}
+					updatedBtObj, err := btc.ds.GetBackupTarget(btObj.Name)
+					if err != nil {
+						log.WithError(err).Errorf("Failed to get new backup target %s", btObj.Name)
+						return err
+					}
+					updatedBtObj.Spec.Default = false
+					if _, err := btc.ds.UpdateBackupTarget(updatedBtObj); err != nil {
+						log.WithError(err).Debugf("Failed to update backup target %s spec", btObj.Name)
+						return err
```

Comment:
If we take this path, it leaves us with no default target. That's better than having two defaults, I suppose.  And it can be fixed by trying again to modify just that setting?

I wonder whether it would be easier to have the zeroth item in the array of targets be the default.  It's changed by reordering the array, but then there always is one, and writing the array can be atomic.